src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs

```diff
@@ -90,18 +76,39 @@ static MsQuicApi()
                         if (StatusSucceeded(apiTable->GetParam(null, QUIC_PARAM_GLOBAL_LIBRARY_VERSION, &size, libVersion)))
                         {
                             var version = new Version((int)libVersion[0], (int)libVersion[1], (int)libVersion[2], (int)libVersion[3]);
-                            if (version >= MsQuicVersion)
-                            {
-                                Api = new MsQuicApi(apiTable);
-                                IsQuicSupported = true;
-                            }
-                            else
+                            if (version < MsQuicVersion)
                             {
                                 if (NetEventSource.Log.IsEnabled())
                                 {
                                     NetEventSource.Info(null, $"Incompatible MsQuic library version '{version}', expecting '{MsQuicVersion}'");
                                 }
+                                return;
+                            }
+
+                            // Assume SChanel is being used on windows and query for the actual provider from the library
+                            QUIC_TLS_PROVIDER provider = OperatingSystem.IsWindows() ? QUIC_TLS_PROVIDER.SCHANNEL : QUIC_TLS_PROVIDER.OPENSSL;
+                            size = sizeof(QUIC_TLS_PROVIDER);
+                            apiTable->GetParam(null, QUIC_PARAM_GLOBAL_TLS_PROVIDER, &size, &provider);
```

Comment:
Do we expect that the provider will not change if there is MsQuic version that does not support the query?

Reply:
The question was really about running on older msquic versions. I don't know if this was added in/prior to 2.0 or if that is more recent addition.

Reply:
I think it's fine to assume this and state that MsQuic+OpenSSL is only supported on Windows for MsQuic v2+.